app/views/user_batches/form.html.erb

```diff
@@ -1,7 +1,33 @@
 <%= elmo_form_for(@user_batch, html: {multipart: true}) do |f| %>
   <p class="instructions"><%= formatted_instructions %></p>
-  <%= f.field(:file, type: :file, accept: ".xlsx,.csv", required: true) %>
+
+  <div id="file-uploader">
+    <%=hidden_field_tag(:temp_file_path, '', id: "temp_file_path")%>
+    <%=hidden_field_tag(:original_filename, '', id: "original_filename")%>
+    <div id="dropzone-file-uploader" class="dropzone">
+    </div>
+    <div class="dz-error-msg"></div>
+  </div>
   <div class="submit-buttons">
     <%= f.submit(t("common.import"), class: "btn btn-primary") %>
   </div>
 <% end %>
+
+<%= javascript_doc_ready do %>
+  ELMO.fileUploaderManager = new ELMO.Views.FileUploaderManager({el: "#new_user_batch"});
+  options = <%=json(
+    el: "#file-uploader",
+    zoneId: "#dropzone-file-uploader",
+    genericThumbPath: nil,
+    postPath: "#{user_batches_path}/upload",
+    previewTemplate: render("layouts/dropzone_preview"),
+    paramName: "userbatch",
+    acceptedFileFormats: ".xlsx, .csv",
+    metaFields: {
+      "tempFilePath" => "input#temp_file_path",
+      "originalFilename" => "input#original_filename"
```

Comment:
I see what is happening here and this is a clever solution but it's is a bit unconventional. Let's talk about how to make this cleaner.